packages/language-server/test/check.test.ts

```diff
@@ -0,0 +1,37 @@
+import { AstroCheck } from '../src/check';
+import { DiagnosticSeverity, Range } from 'vscode-languageserver-types';
+import ts from 'typescript/lib/tsserverlibrary';
+import { pathToUrl } from '../src/utils';
+import { expect } from 'chai';
+import { join } from 'path';
+import { harmonizeNewLines } from './utils';
+
+describe('astro check', async () => {
+	it('should check astro projects', async () => {
+		const path = join(__dirname, '/fixtures/astro-check/hasError.astro');
+		const checker = new AstroCheck(path, require.resolve('typescript/lib/tsserverlibrary.js'));
+
+		checker.upsertDocument({
+			uri: pathToUrl(path),
+			text: harmonizeNewLines(ts.sys.readFile(path) || ''),
+		});
+
+		const diagnostics = await checker.getDiagnostics();
+		expect(diagnostics).to.deep.equal([
+			{
+				diagnostics: [
+					{
+						code: 2304,
+						message: "Cannot find name 'doesntExist'.",
+						range: Range.create(1, 1, 1, 12),
+						severity: DiagnosticSeverity.Error,
+						source: 'ts',
+						tags: [],
+					},
+				],
+				filePath: new URL(pathToUrl(path)).pathname,
```

Comment:
This is necessary because Astro check currently handle paths the wrong way. It needs a minor to fix so I'll do that in the future

Reply:
This will break on Windows, IIRC. Are you looking for `pathToFileURL()`?

Reply:
This is actually exactly so the test pass on Windows haha. The language-server's part of `astro check` returns paths the wrong way, the Astro part of `astro check` fixes them back up for display.